File: shared/frontend-client/src/types/points.ts
export type PointLedgerType = "earn" | "spend" | "adjust";

export interface PointBalanceRead {
  user_id: string;
  points_balance: number;
}

export interface PointLedgerRead {
  id: number;
  user_id: string;
  order_id: number | null;
  type: PointLedgerType;
  amount: number;
  balance_after: number;
  created_at: string;
}

File: backend/app/app/schemas/points.py
from datetime import datetime

from pydantic import BaseModel

from app.models.enums import PointLedgerType


class PointBalanceRead(BaseModel):
    user_id: str
    points_balance: int


class PointLedgerRead(BaseModel):
    id: int
    user_id: str
    order_id: int | None = None
    type: PointLedgerType
    amount: int
    balance_after: int
    created_at: datetime
